### ISSUE TITLE
Indent errors on comments inside if blocks

### ISSUE THREAD

**Issue by paulschreiber** [WordPress-Coding-Standards/WordPress-Coding-Standards#288]
Suppose I have some code like this

    if ( true ) {
      $foo = 1;
    // remember that the shoes are green
    } else {
      $shoes->size();
    }

I'll get this error about the comment:

    Line indented incorrectly; expected at least 2 tabs, found 1

Since the comment describes what's going on with the else block, having it indented only one tab in makes sense.

**Comment by westonruter** [comment]
I like that.

**Comment by paulschreiber** [comment]
I disagree with Gary. You can't visually differentiate comments about the block as a whole from comments about the first time inside the condition. I also find his proposal harder to read -- it's easier for the comments to get lost when they're indented inside the control structure.

@aliso also flagged this in a code review (she doesn't like it either).

**Comment by JDGrimes** [comment]
Sometimes I do this:

```php
if ( ... ) {
    // - If something, then do something

    // Comment about next line.
    ...
} else {
    // - Otherwise, do something else

    // Comment about next line
    ...
}
```

Using `// -` for comments that act like a heading for an entire block helps them to stand out a bit more. But I think it is usually pretty obvious, since the comment after the `if` will almost always start with `If...` and the comment after the `else` will almost always start with `Otherwise...`

**Comment by westonruter** [comment]
To allow for this style:

```php
// If something do something.
if ( $this ) {
    ...
// If something else, do something else.
} elseif ( $this_other_thing ) {
```

What we need to do is update the sniff to allow a comment to be indented one level lower than it would normally be allowed if it immediately precedes a closing brace which is then followed by `else if` or `else`.

**Comment by GaryJones** [comment]
> You can't visually differentiate comments about the block as a whole from comments about the first time inside the condition.

You can still comment the whole block:

~~~php
// Some comment about why we have a if/else here
if ( ... ) {
    // Some comment about this block
    ...
} else {
    // Another comment about this block
    ...
}
~~~

**Comment by GaryJones** [comment]
Going back to the original post, I still think this is a valid error. The code (and comment) inside that `if { }` block is not consistently indented. The fact the developer might be mentally meaning a final comment in a block it to convey information about the following block, is not applicable or testable here.

**Comment by jrfnl** [comment]
Wouldn't life be so much easier if we'd just have `else` of a new line ? :innocent: :heart_eyes: :stuck_out_tongue_closed_eyes:
